app/models/Status.scala

```diff
@@ -17,38 +17,107 @@
+      case Amended                      => JsString("07")
+      case Released                     => JsString("08")
+      case Cleared                      => JsString("09")
+      case Cancelled                    => JsString("10")
+      case CustomsPositionGranted       => JsString("1139")
+      case CustomsPositionDenied        => JsString("1141")
+      case GoodsHaveExitedTheCommunity  => JsString("16")
+      case DeclarationHandledExternally => JsString("17")
+      case AwaitingExitResults          => JsString("18")
     }
   }
 
+  def retrieveFromResponse(response: Response): Status =
+    response.functionCode match {
+      case "01"                                                                 => Accepted
+      case "02"                                                                 => Received
+      case "03"                                                                 => Rejected
+      case "05"                                                                 => UndergoingPhysicalCheck
+      case "06"                                                                 => AdditionalDocumentsRequired
+      case "07"                                                                 => Amended
+      case "08"                                                                 => Released
+      case "09"                                                                 => Cleared
+      case "10"                                                                 => Cancelled
+      case "11" if response.status.headOption.flatMap(_.nameCode) == Some("39") => CustomsPositionGranted
+      case "11" if response.status.headOption.flatMap(_.nameCode) == Some("41") => CustomsPositionDenied
+      case "16"                                                                 => GoodsHaveExitedTheCommunity
+      case "17"                                                                 => DeclarationHandledExternally
+      case "18"                                                                 => AwaitingExitResults
+      case _                                                                    => throw new Exception("Incorrect status")
+    }
 }
 
 case object Accepted extends Status
 
-case object Arrived extends Status
+case object Received extends Status
+
+case object Rejected extends Status
 
-case object Departed extends Status
+case object UndergoingPhysicalCheck extends Status {
+  override def toString(): String = "Undergoing Physical Check"
+}
+
+case object AdditionalDocumentsRequired extends Status {
+  override def toString(): String = "Additional Documents Required"
+}
 
-case object PreLodged extends Status
+case object Amended extends Status
 
-case object Query extends Status
+case object Released extends Status
+
+case object Cleared extends Status
+
+case object Cancelled extends Status
+
+case object CustomsPositionGranted extends Status {
+  override def toString(): String = "Customs Position Granted"
```

Comment:
move these into ENUM please

Reply:
Why?

Reply:
https://medium.com/@yuriigorbylov/scala-enumerations-hell-5bdba2c1216

If you can I can only move it to `java.util.Enum`, but I don't see any pros of it